[CI] GenerativeIT test {feature:BASELINE} failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #12496 / amazonlinux-2023_platform-support-aws](https://gradle-enterprise.elastic.co/s/eo6cfstxgoeym)
- [elasticsearch-periodic-platform-support #12496 / amazonlinux-2023_platform-support-aws](https://gradle-enterprise.elastic.co/s/k5bsghkskhp5k)
- [elasticsearch-pull-request #163595 / part-3](https://gradle-enterprise.elastic.co/s/vcolcy3hprk4s)
- [elasticsearch-pull-request #163474 / part-3](https://gradle-enterprise.elastic.co/s/bn3q35tberfq2)
- [elasticsearch-pull-request #163265 / part-3](https://gradle-enterprise.elastic.co/s/mbp5pwrerif76)
- [elasticsearch-pull-request #163265 / part-3](https://gradle-enterprise.elastic.co/s/bx7ovffgljvm4)
- [elasticsearch-periodic #12518 / encryption-at-rest](https://gradle-enterprise.elastic.co/s/khauq4y6vhjqg)
- [elasticsearch-pull-request #163222 / releasetestcheckesqlqa1_release-tests](https://gradle-enterprise.elastic.co/s/alh5ciz6p5yua)
- [elasticsearch-pull-request #163222 / releasetestcheckesqlqa1_release-tests](https://gradle-enterprise.elastic.co/s/s7grxe6x2wtyg)
- [elasticsearch-periodic-platform-support #12491 / almalinux-8-aarch64_checkpart3_platform-support-arm](https://gradle-enterprise.elastic.co/s/ebedevaexc22g)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:single-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.single_node.GenerativeIT.test {feature:BASELINE}" -Dtests.seed=AD152E993F1737E9 -Dtests.locale=uz-Arab -Dtests.timezone=America/Indiana/Indianapolis -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.single_node.GenerativeIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test%20%7Bfeature%3ABASELINE%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: query: ROW vynsGQGk = false, HxRlXMuhDQua = 7795888188632723545, HxRlXMuhDQua = true, kcRiZNVPVkeZ = null, AaZtPmBoThI = -762667968808990777, IivLVikXsL = false, vMhZHJilvVd = 2082369186, ejTfSoFNLF = [4545036208826450905, 834246700172555022, 9007673597597798300, -5912417176537982735, 3600792600472465248, 2723692927513481198, -75246330963104039], vynsGQGk = -637366285 | eval  zAZAcUQKo = null, FeNwcKnBYC = null, vjySggsvn = "otpnhqbq", PnIHFsQjPl = vMhZHJilvVd, PSTpKtaj = vMhZHJilvVd, ejTfSoFNLF = floor(ejTfSoFNLF), `vynsGQGk` = "lyHvMD", JLNKPaUCEBpx = 428477931005662783, ZLLjtestQuy = mv_difference(mv_dedupe(least(vMhZHJilvVd)), case(mv_intersection(HxRlXMuhDQua, HxRlXMuhDQua) OR mv_union(HxRlXMuhDQua, IivLVikXsL), round(vMhZHJilvVd))), jqYeKCOIgKhw = least(vMhZHJilvVd, -6, 65)
exception: method [POST], host [http://[::1]:39221], URI [/_query?pretty=true&error_trace=true], status line [HTTP/1.1 500 Internal Server Error]
Warnings: [No limit defined, adding d
[truncated]
```

**Issue Reasons:**
- [main] 2 consecutive failures in test test {feature:BASELINE}
- [main] 2 consecutive failures in step amazonlinux-2023_platform-support-aws
- [main] 1 consecutive failures in step encryption-at-rest
- [main] 2 consecutive failures in step releasetestcheckesqlqa_release-tests
- [main] 1 consecutive failures in pipeline elasticsearch-periodic-platform-support
- [main] 21 failures in test test {feature:BASELINE} (4.5% fail rate in 468 executions)
- [main] 2 failures in step amazonlinux-2023_platform-support-aws (66.7% fail rate in 3 executions)
- [main] 8 failures in step part-3 (3.4% fail rate in 238 executions)
- [main] 7 failures in step releasetestcheckesqlqa1_release-tests (8.0% fail rate in 88 executions)
- [main] 2 failures in step releasetestcheckesqlqa_release-tests (100.0% fail rate in 2 executions)
- [main] 2 failures in pipeline elasticsearch-periodic-platform-support (100.0% fail rate in 2 executions)
- [main] 9 failures in pipeline elasticsearch-pull-request (3.8% fail rate in 238 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -421,9 +421,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.qa.multi_node.GenerativeIT
   method: test {feature:SUBQUERIES}
   issue: https://github.com/elastic/elasticsearch/issues/149681
-- class: org.elasticsearch.xpack.esql.qa.multi_node.GenerativeIT
-  method: test {feature:BASELINE}
-  issue: https://github.com/elastic/elasticsearch/issues/149746
 - class: org.elasticsearch.datastreams.lifecycle.ExplainDataStreamLifecycleIT
   method: testExplainFailuresLifecycle
   issue: https://github.com/elastic/elasticsearch/issues/151602
@@ -832,6 +829,4 @@ tests:
 - class: org.elasticsearch.xpack.esql.CsvIT
   method: test {csv-spec:inlinestats.groupByMultipleRenamedColumns_AndComplexEval}
   issue: https://github.com/elastic/elasticsearch/issues/153752
-- class: org.elasticsearch.xpack.esql.qa.single_node.GenerativeIT
-  method: test {feature:BASELINE}
-  issue: https://github.com/elastic/elasticsearch/issues/149746
+

--- a/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/GenerativeRestTest.java
+++ b/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/GenerativeRestTest.java
@@ -160,7 +160,7 @@ public abstract class GenerativeRestTest extends ESRestTestCase implements Query
         "Field \\[.*\\] of type \\[.*\\] does not support match.* queries",
 
         // https://github.com/elastic/elasticsearch/issues/153622
-        "ImmutableCollections\\$ListN cannot be cast to class org.apache.lucene.util.BytesRef",
+        "ImmutableCollections\\$ListN cannot be cast to class .*",
 
         // repeat() returns validation error when the Number parameter is a negative foldable
         "Number parameter cannot be negative, found \\[",
@@ -862,6 +862,7 @@ public abstract class GenerativeRestTest extends ESRestTestCase implements Query
 
     private static List<Column> handleRenameIndexMapped(List<Column> newSchema, Map<String, Boolean> prevMapped, String commandString) {
         Map<String, Boolean> mapped = new HashMap<>(prevMapped);
+        Set<String> renamed = new HashSet<>();
         String body = commandString.replaceFirst("(?i)^\\s*\\|\\s*rename\\s+", "");
         for (String pair : body.split(",")) {
             Matcher m = RENAME_PAIR_PATTERN.matcher(pair);
@@ -871,11 +872,16 @@ public abstract class GenerativeRestTest extends ESRestTestCase implements Query
                 boolean wasMapped = mapped.getOrDefault(oldName, false);
                 mapped.remove(oldName);
                 mapped.put(newName, wasMapped);
+                renamed.add(newName);
             }
         }
+        // After RENAME, the new name is a ReferenceAttribute, not a FieldAttribute.
+        // ES|QL's Match.isRuntimeSearch() uses fieldAsFieldAttribute() which cannot resolve renamed
+        // fields → options are not allowed on them. Mark renamed targets as non-index-mapped so the
+        // generator won't try to use them in match() with options.
         return newSchema.stream().map(col -> {
-            Boolean isMapped = mapped.get(col.name());
-            return new Column(col.name(), col.type(), col.originalTypes(), isMapped != null && isMapped);
+            boolean isMapped = mapped.getOrDefault(col.name(), false) && renamed.contains(col.name()) == false;
+            return new Column(col.name(), col.type(), col.originalTypes(), isMapped);
         }).toList();
     }
 

--- a/x-pack/plugin/esql/qa/testFixtures/src/main/java/org/elasticsearch/xpack/esql/generator/function/SpecialFunctionGeneratorRegistry.java
+++ b/x-pack/plugin/esql/qa/testFixtures/src/main/java/org/elasticsearch/xpack/esql/generator/function/SpecialFunctionGeneratorRegistry.java
@@ -38,8 +38,17 @@ public final class SpecialFunctionGeneratorRegistry {
     /** Algorithm names accepted by the {@code hash} function. */
     private static final String[] HASH_ALGORITHMS = { "MD5", "SHA-256", "SHA-384", "SHA-512" };
 
-    /** Time-unit strings accepted by {@code date_diff} and {@code date_unit_count}. */
+    /** Time-unit strings accepted by {@code date_diff}. */
     private static final String[] DATE_DIFF_UNITS = { "second", "minute", "hour", "day", "week", "month", "year" };
+
+    /**
+     * Valid {@code from_unit} values for {@code date_unit_count}.
+     * The function's {@code count()} method only handles variable-length units (day/week/month/year)
+     * as {@code from_unit}; fixed-length units (second/minute/hour) reach an {@code IllegalArgumentException}
+     * unless they cancel with the {@code to_unit} via {@code constNanos}. Using only calendar units
+     * here avoids that runtime error.
+     */
+    private static final String[] DATE_UNIT_COUNT_FROM_UNITS = { "day", "week", "month", "year" };
 
     /** Date-part names accepted by {@code date_extract}. */
     private static final String[] DATE_EXTRACT_PARTS = {
@@ -105,10 +114,13 @@ public final class SpecialFunctionGeneratorRegistry {
         });
 
         // date_unit_count(to_unit, from_unit, date): both unit params must be literals.
+        // from_unit must be a calendar (variable-length) unit: day/week/month/year.
+        // Fixed-length units (second/minute/hour) as from_unit reach an unsupported code path
+        // and throw IllegalArgumentException at runtime when the to_unit is also variable-length.
         r.put("date_unit_count", (name, sig, cols, unmapped, depth, recurse) -> {
             String date = recurse.recurse(sig.params().get(2).type(), cols, unmapped, depth - 1);
             if (date == null) return null;
-            return name + "(\"" + randomFrom(DATE_DIFF_UNITS) + "\", \"" + randomFrom(DATE_DIFF_UNITS) + "\", " + date + ")";
+            return name + "(\"" + randomFrom(DATE_DIFF_UNITS) + "\", \"" + randomFrom(DATE_UNIT_COUNT_FROM_UNITS) + "\", " + date + ")";
         });
 
         // ---- MV functions ---------------------------------------------------

